### DISCUSSION TITLE
Javascript, to load or not to load inside template files, that's the question....

### DISCUSSION THREAD

**Discussion by rica-carv** [e107inc/e107#5258] [General]
I have a plugin with a template design that has some jscript to make it more "user friendly", but the same purpose can be anchieved without any jscript. RIght now the jscript is loaded with e107::js at the logic file, ie plugin,php, but i was wondering, since jscript is a client side scripting, and perhaps has more to do with presentation than logic, couldn't or shouldn't jscript be template dependent and be loaded only if the template explicity it? It could even be jscripts code that the template designer wants to use, and there's no need to mess with the plugin core php file...
Or there's no need for the core php file load jscripts that the template might or not might use....

Just my thoughts....

**Comment by Jimmi08** [comment]
Featurebox uses this approach - each template key can have its own javascript.  This is nice for the initialization of some scripts too. But from this approach, it was moved to behaviors.  Actually, gallery plugin has an issue with the initialization of photo gallery   and I am not able to fix it. It is too complicated now.

I am using for this plugin preference to load only needed JS if it depends on the template or so.   For example, the theme can use the same script as your plugin template and it would be loaded 2x. 

And I think you can use e107::js in the template file too.

**Reply by rica-carv** [reply]
> And I think you can use e107::js in the template file too.

Yes you can, i'm using it and abusing it on my plugin templates....
But somewhere in time, i realized there's a e_header.php file i can use to insert all css and js files on the top of page, at page generation... but using this way, it's gonna be a real mess real quick... Imagine you have a js file load in template, in e_header, in the core plugin file....
Yikes.... You'll end up with your page cluttered with js files for nothing....

So ithink it should be some guidelines for not messing all things up...

@Jimmi08 I don't use Featurebox, but if you want me to have a look, just let me know what you want... right now i'm struggling with js on my plugin also... :wink:

**Reply by Vodhin** [reply]
JavaScript runs on a user's device. In the old days, when a computer's RAM was tiny, it made sense to try to keep you JS small. Today you got plenty of room on nearly every device out there and modern browsers and hardware know how to manage resources nicely. So, don't worry too much about writing large JavaScript files. I have one main one for ADMINs with 10,000 + lines of code that weighs in at 366KB. Runs fine on my 1998 Dell Windows XP desktop and whatever version of Firefox can run on that...
